infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -396,9 +419,28 @@ private void stopLogReplication() {
     public void processLockAcquire() {
         log.debug("Process lock acquire event");
         isLeader = true;
+
+        // Regardless of the role of this node, on leadership acquisition
+        // we shall query the topology provider to get the most up to date
+        // topologyConfigId, which might have changed during our time as non-leaders
+        // and for which we might have missed the change notification and update it on
+        // the sink manager so the value is cached to filter messages upon receive.
+        fetchTopologyFromClusterManager();
+        updateTopologyConfigId(topologyDescriptor.getTopologyConfigId());
```

Comment:
Just doing updateTopologyConfigID is not enough. While process a topology change event, the discovery service does more work refer to processTopologyChangeNotification. 

While combining two events topology change and leadership change together, it is a bit complex.